extensions-core/namespace-lookup/src/main/java/io/druid/server/namespace/cache/OffHeapNamespaceExtractionCacheManager.java

```diff
@@ -119,6 +119,7 @@ protected boolean swapAndClearCache(String namespaceKey, String cacheKey)
 
       final String priorCache = currentNamespaceCache.put(namespaceKey, swapCacheKey);
       if (priorCache != null) {
+        // TODO: resolve what happens here if query is actively going on
```

Comment:
Eventually, yes. Unless you have some sort of insight, though, this is not straight forward. The issue is that queries do not have a concept of resources associated with them. There are some runners which wrap resources, but there isn't a general concept of query resources that need to be accounted for.

There are quite a few resources that get tied to a particular query though, so it might be worthwhile to add such a convention.